ui/client/datasets/DataTransformation/useElwoodData.js

```diff
@@ -9,6 +9,7 @@ const useElwoodData = ({
   generateArgs,
   jobString,
   cleanupRef,
+  onBackendFailure
```

Comment:
`onBackendFailure` needs to be added as a dependency to the useEffect. I disabled the linter for the dependency array here because I didn't think I should be adding the ref to it, but on further exploration I believe it should be totally safe to add the ref without triggering any extra useEffect updates. So we can remove  `eslint-disable-next-line react-hooks/exhaustive-deps` from line 116 and add `cleanupRef` (and `onBackendFailure`).